File: backend/src/main/java/com/example/demo/bean/Constraint.java
package com.example.demo.bean;

import java.util.List;

public class Constraint extends ProblemDiagramLine {
	private int constraint_no;	//约束编号
	private String constraint_name;	//约束名称
	private String constraint_description;	//约束详情
	private String constraint_from;	
	private String constraint_to;
	private int constraint_x1;	//位置信息
	private int constraint_y1;
	private int constraint_x2;
	private int constraint_y2;
	private List<RequirementPhenomenon> phenomenonList;	//现象列表
	public int getNo() {
		return constraint_no;
	}
	public void setNo(int constraint_no) {
		this.constraint_no = constraint_no;
	}
	public String getName() {
		return constraint_name;
	}
	public void setName(String constraint_name) {
		this.constraint_name = constraint_name;
	}
	public String getDescription() {
		return constraint_description;
	}
	public void setDescription(String constraint_description) {
		this.constraint_description = constraint_description;
	}
	public String getFrom() {
		return constraint_from;
	}
	public void setFrom(String constraint_from) {
		this.constraint_from = constraint_from;
	}
	public String getTo() {
		return constraint_to;
	}
	public void setTo(String constraint_to) {
		this.constraint_to = constraint_to;
	}
	public int getX1() {
		return constraint_x1;
	}
	public void setX1(int constraint_x1) {
		this.constraint_x1 = constraint_x1;
	}
	public int getY1() {
		return constraint_y1;
	}
	public void setY1(int constraint_y1) {
		this.constraint_y1 = constraint_y1;
	}
	public int getX2() {
		return constraint_x2;
	}
	public void setX2(int constraint_x2) {
		this.constraint_x2 = constraint_x2;
	}
	public int getY2() {
		return constraint_y2;
	}
	public void setY2(int constraint_y2) {
		this.constraint_y2 = constraint_y2;
	}
	
	public int getConstraint_no() {
		return constraint_no;
	}
	public void setConstraint_no(int constraint_no) {
		this.constraint_no = constraint_no;
	}
	public String getConstraint_name() {
		return constraint_name;
	}
	public void setConstraint_name(String constraint_name) {
		this.constraint_name = constraint_name;
	}
	public String getConstraint_description() {
		return constraint_description;
	}
	public void setConstraint_description(String constraint_description) {
		this.constraint_description = constraint_description;
	}
	public String getConstraint_from() {
		return constraint_from;
	}
	public void setConstraint_from(String constraint_from) {
		this.constraint_from = constraint_from;
	}
	public String getConstraint_to() {
		return constraint_to;
	}
	public void setConstraint_to(String constraint_to) {
		this.constraint_to = constraint_to;
	}
	public int getConstraint_x1() {
		return constraint_x1;
	}
	public void setConstraint_x1(int constraint_x1) {
		this.constraint_x1 = constraint_x1;
	}
	public int getConstraint_y1() {
		return constraint_y1;
	}
	public void setConstraint_y1(int constraint_y1) {
		this.constraint_y1 = constraint_y1;
	}
	public int getConstraint_x2() {
		return constraint_x2;
	}
	public void setConstraint_x2(int constraint_x2) {
		this.constraint_x2 = constraint_x2;
	}
	public int getConstraint_y2() {
		return constraint_y2;
	}
	public void setConstraint_y2(int constraint_y2) {
		this.constraint_y2 = constraint_y2;
	}
	public List<RequirementPhenomenon> getPhenomenonList() {
		return phenomenonList;
	}
	public void setPhenomenonList(List<RequirementPhenomenon> phenomenonList) {
		this.phenomenonList = phenomenonList;
	}
}

File: frontend/src/app/entity/Constraint.ts
import { RequirementPhenomenon } from './RequirementPhenomenon';
import { Line } from './Line'
export class Constraint extends Line{
	no: number;	
	name: string;	
	description: string;	
	from: string;	
	to: string;
	x1: number;	
	y1: number;
	x2: number;
	y2: number;
	phenomenonList: RequirementPhenomenon[];
	static newConstraint(no, name, description,from,to,phenomenonList,x1,y1,x2,y2){
		let con = new Constraint()
		con.no=no
		con.name=name
		con.description=description
		con.from=from
		con.to=to
		if(phenomenonList)
			con.phenomenonList=phenomenonList
		else 
			con.phenomenonList = []
		con.x1=x1
		con.y1=y1
		con.x2=x2
		con.y2=y2
		return con
	}
	static newConstraintWithOld(old, phenomenonList, description){
		let con = new Constraint()
		con.no = old.no
		con.name = old.name
		con.from = old.from
		con.to = old.to
		con.x1 = old.x1
		con.y1 = old.y1
		con.x2 = old.x2
		con.y2 = old.y2
		con.description=description
		if(phenomenonList)
			con.phenomenonList=phenomenonList
		else 
			con.phenomenonList = []
		return con
	}
	static copyConstraint(old){
		let con = new Constraint()
		con.no = old.no
		con.name = old.name
		con.from = old.from
		con.to = old.to
		con.x1 = old.x1
		con.y1 = old.y1
		con.x2 = old.x2
		con.y2 = old.y2
		con.description = old.description
		if(old.phenomenonList)
			con.phenomenonList=old.phenomenonList
		else 
			con.phenomenonList = []

		return con
	}
	getNo(){return this.no}	
	setNo(no){this.no=no}

	getName(){return this.name}	
	setName(name){this.name=name}

	getDescription(){ return this.description}
	setDescription(description){ this.description=description}

	getFrom(){return this.from}
	setFrom(from){this.from=from}

	getTo(){return this.to}
	setTo(to){ this.to=to}

	getX1(){return this.x1}
	setX1(x1){ this.x1=x1}

	getY1(){return this.y1}
	setY1(y1){ this.y1=y1}

	getX2(){return this.x2}
	setX2(x2){ this.x2=x2}

	getY2(){return this.y2}
	setY2(y2){ this.y2=y2}
	getPhenomenonList(){return this.phenomenonList}
	setPhenomenonList(phenomenonList){ this.phenomenonList=phenomenonList}
}